source/common/config/json_utility.h

```diff
@@ -50,3 +50,13 @@
           1000 * (json).getInteger(#field_name "_s")));                                            \
     }                                                                                              \
   } while (0)
+
+// Set a google.protobuf.Duration field in a protobuf message with the specified field's
+// milliseconds value from a JSON object if the field is set in the JSON object.
+#define JSON_UTIL_SET_DURATION_FROM_FIELD(json, message, dst_field, src_field)                     \
```

Comment:
Nit: probably should define `JSON_UTIL_SET_DURATION` in terms of this macro now.

Reply:
can I do this in next PR for redis?

Reply:
Yes that's fine. Please do in the next PR.